test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataInCookiesTest.cs

```diff
@@ -135,5 +135,35 @@ public TempDataInCookiesTest(MvcTestFixture<BasicWebSite.StartupWithCookieTempDa
             Assert.NotNull(setCookieHeader.Expires);
             Assert.True(setCookieHeader.Expires < DateTimeOffset.Now); // expired cookie
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task CookieTempDataProviderCookie_SetsSecureAttributeOnCookie_OnlyIfRequestIsSecure(bool secureRequest)
```

Comment:
A functional test for this seems like a bit of overkill doesn't it? Couldn't you change one of the other tests to pass in `https`?